rust/src/write.rs

```diff
@@ -388,37 +376,93 @@ impl<'a, W: Write + Seek> Writer<'a, W> {
         Ok(())
     }
 
-    pub fn attach(&mut self, attachment: &Attachment) -> McapResult<()> {
-        let header = records::AttachmentHeader {
-            log_time: attachment.log_time,
-            create_time: attachment.create_time,
-            name: attachment.name.clone(),
-            media_type: attachment.media_type.clone(),
+    /// Start writing an attachment.
+    ///
+    /// This is a low level API. It is recommended to use [`Self::attach`] instead.
+    ///
+    /// To start writing an attachment call this method with the [`AttachmentHeader`] as well as
+    /// the length of the attachment in bytes. It is important this length is exact otherwise the
+    /// writer will be left in an error state.
+    ///
+    /// This call should be followed by one or more calls to [`Self::put_attachment_bytes`].
+    ///
+    /// Once all attachment bytes have been written the attachment must be completed with a call to
+    /// [`Self::finish_attachment`]. Failing to finish the attachment will leave the write in an
+    /// error state.
```

Comment:
would be good to include an example which shows how to copy a file to an attachment correctly.